k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -230,7 +231,12 @@ private[k8s] class NsObjectResource[O <: KubeObject: TypeReference, W <: Watch[O
       None,
       retryIndefinitely = true,
       watch = true
-    ).map { obj =>
-      (Seq(od.toWatch(obj)), obj.metadata.flatMap(_.resourceVersion))
+    ).map { maybeObj =>
+      (maybeObj.toSeq.flatMap { obj => Seq(od.toWatch(obj)) },
+        for {
+          obj <- maybeObj
+          meta <- obj.metadata
+          version <- meta.resourceVersion
+        } yield version)
```

Comment:
this would be easier to read with temp vals

```scala
val w = maybeObj.toSeq.flatMap { obj => Seq(od.toWatch(obj)) }
val v = for {
  obj <- maybeObj
  meta <- obj.metadata
  version <- meta.resourceVersion
} yield version
(w, v)
```